src/components/custom-field-input-date/custom-field-input-date.jsx

```diff
@@ -68,12 +76,17 @@ const CustomFieldInputDate = forwardRef(function CustomFieldInputDate(props, ref
 
   useImperativeHandle(ref, () => ({
     get dirty() {
-      return props.value !== this.value;
+      return convertToFormat(props.value, 'yyyy-mm-dd') !== this.value;
```

Comment:
The tests for this are `xit`ed at the moment. When I tried writing them, I was getting hung up on trying to type into the date input. Maybe we can give it another go or try setting value without typing.